contracts/test/forkMainnet/L2Deposit/Deposit.t.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.7.6;
+pragma abicoder v2;
+
+import "contracts/interfaces/IL2Deposit.sol";
+import "contracts-test/forkMainnet/L2Deposit/Setup.t.sol";
+import "contracts-test/utils/BalanceSnapshot.sol";
+
+contract TestL2DepositTopUp is TestL2Deposit {
+    using BalanceSnapshot for BalanceSnapshot.Snapshot;
+
+    uint256 arbMaxSubmissionCost = 1e18;
+    uint256 arbMaxGas = 1e6;
+    uint256 arbGasPriceBid = 1e6;
+
+    function testCannotDepositIfExpired() public {
+        // overwrite expiry
+        DEFAULT_DEPOSIT.expiry = block.timestamp;
+        // sig is not relevent in this case
+        bytes memory sig = bytes("");
+        bytes memory payload = abi.encodeWithSelector(L2Deposit.deposit.selector, IL2Deposit.DepositParams(DEFAULT_DEPOSIT, sig));
+
+        vm.expectRevert("L2Deposit: Deposit is expired");
+        userProxy.toL2Deposit(payload);
+    }
+
+    function testCannotDepositWithInvalidSig() public {
+        // compose payload with signature from bob
+        bytes memory sig = _signDeposit(bobPrivateKey, DEFAULT_DEPOSIT);
+        bytes memory payload = abi.encodeWithSelector(L2Deposit.deposit.selector, IL2Deposit.DepositParams(DEFAULT_DEPOSIT, sig));
+
+        vm.expectRevert("L2Deposit: Invalid deposit signature");
+        userProxy.toL2Deposit(payload);
+    }
+
+    function testArbitrumDeposit() public {
+        BalanceSnapshot.Snapshot memory userL1TokenBal = BalanceSnapshot.take(user, DEFAULT_DEPOSIT.l1TokenAddr);
+
+        // overwrite deposit data with encoded arbitrum specific params
+        DEFAULT_DEPOSIT.data = abi.encode(arbMaxSubmissionCost, arbMaxGas, arbGasPriceBid);
+
+        // compose payload with signature
+        bytes memory sig = _signDeposit(userPrivateKey, DEFAULT_DEPOSIT);
+        bytes memory payload = abi.encodeWithSelector(L2Deposit.deposit.selector, IL2Deposit.DepositParams(DEFAULT_DEPOSIT, sig));
+        uint256 callValue = arbMaxSubmissionCost + (arbMaxGas * arbGasPriceBid);
+
+        // Response from bridge is unknown
+        vm.expectEmit(true, true, true, false);
+        emit Deposited(
+            DEFAULT_DEPOSIT.l2Identifier,
+            DEFAULT_DEPOSIT.l1TokenAddr,
+            DEFAULT_DEPOSIT.l2TokenAddr,
+            DEFAULT_DEPOSIT.sender,
+            DEFAULT_DEPOSIT.recipient,
+            DEFAULT_DEPOSIT.amount,
+            DEFAULT_DEPOSIT.data,
+            bytes("")
+        );
```

Comment:
The bridge response will be `abi.encode(seqNum)`, and the next seq num could be obtained by `IArbitrumL1Bridge.messageCount`

You could evaluate if it is worth to test it exactly.

Reply:
Old test reference: https://github.com/consenlabs/tokenlon-contracts-private/blob/b749397a554b1f9b46ade7f45c05071514e50c63/archived/test/unit_tests/L2Deposit.ts#L392-L395

Reply:
I found the returned value is `delayedMessageCount` in Bridge contract. Fixed in 8050eb3.